app/models/material.rb

```diff
@@ -8,4 +8,12 @@ class Material < ApplicationRecord
   validates :material_type, presence: true
   validates :status, presence: true
   validates :employee_id, presence: false
+  validates :serial_number, format: { with:/\d{3}\-\d{3}\-\d{3}/}, unless: :flag?
```

Comment:
The `flag?` condition here is problematic because 1.) it's hard to know what's going on here since the name is not descriptive and 2.) since the value of flag is still coming from a params in controller, the controller and model are still linked together. ideally, model logic should just be in model code. 
refactor this validation so that the validation will be based on the value of `material_type` and not `@flag`. [see this for reference](https://guides.rubyonrails.org/active_record_validations.html#using-a-symbol-with-if-and-unless)